### PR TITLE
Replace ethereum_rpc_client with web3.py #690

### PR DESCRIPTION
**Pull request by banasiakadam60**
Signed-off-by: Adam Banasiak <banasiakadam60@gmail.com>

### CODE REVIEW

**Comment by banasiakadam60** [comment]
#690

**Comment by chfast** [comment]
Still some failures in tests.

File: golem/ethereum/client.py [hunk 1]
```diff
@@ -22,43 +21,131 @@ def __init__(self, datadir, nodes=None):
...
+        }
+        return self.web3.eth.filter(obj)
+
+    def get_filter_changes(self, filer_id):
+        """
+        Polling method for a filter, which returns an array of logs which occurred since last poll
+        :param filer_id: the filter id
+        :return: Returns all new entries which occurred since the last call to this method for the given filter_id
+        """
+        if hasattr(filer_id, 'filter_id'):  # in case of passing filter object
```
Comment (chfast) [inline golem/ethereum/client.py]: I guess the new_filler should return the `.filter_id` straight away. Besides you have type in `filer_id`.

File: golem/ethereum/client.py [hunk 2]
```diff
@@ -22,43 +21,131 @@ def __init__(self, datadir, nodes=None):
...
+        or in the case of a contract-creation transaction, the initialisation code
+        :param nonce: Integer of a nonce. This allows to overwrite your own pending transactions that use the same nonce
+        :param block: integer block number, or the string "latest", "earliest" or "pending"
+        :return: The returned data of the call, e.g. a codes functions return value
+        """
+        _from = _from or self.web3.eth.defaultAccount
+        block = block or self.web3.eth.defaultBlock
+
+        obj = {
+            'from': _from,
```
Comment (chfast) [inline golem/ethereum/client.py:104]: What happens if you omit the "from" field?

File: golem/ethereum/node.py [hunk 3]
```diff
@@ -103,6 +103,7 @@ def start(self, rpc, mining=False, nodekey=None, port=None):
             '--datadir', self.datadir,
             '--networkid', '9',
             '--port', str(self.port),
+            '--etherbase', '0x6528d9354356d7f668c75e9ed97f792bf910c8e5',
```
Comment (chfast) [inline golem/ethereum/node.py]: That make no sense.

File: golem/ethereum/paymentmonitor.py [hunk 4]
```diff
@@ -65,6 +65,6 @@ def process_incoming_payments(self):
                              'tx_hash': tx_hash}
             self.__payments.append(payment)
             log.info("Incoming payment: {} -> ({} ETH)".format(
-                     payer.encode('hex'), value / denoms.ether))
+                payer.encode('hex'), value / denoms.ether))
```
Comment (chfast) [inline golem/ethereum/paymentmonitor.py:68]: Avoid not related changes in other files.

**Review by chfast — CHANGES_REQUESTED** [review]

File: golem/ethereum/client.py (hunk 2 above)
Comment (banasiakadam60) [reply golem/ethereum/client.py:104]: Due to documentation it's optional field with default value `web3.eth.defaultAccount`, so I've added `_from = _from or self.web3.eth.defaultAccount`
See: http://web3py.readthedocs.io/en/latest/web3.eth.html `sendTransaction`

File: golem/ethereum/client.py (hunk 1 above)
Comment (banasiakadam60) [reply golem/ethereum/client.py]: OK

File: golem/ethereum/node.py (hunk 3 above)
Comment (banasiakadam60) [reply golem/ethereum/node.py]: Removed

File: golem/ethereum/paymentmonitor.py (hunk 4 above)
Comment (banasiakadam60) [reply golem/ethereum/paymentmonitor.py:68]: Sorry, my IDE automatically changed it and I've accidentally sent this change

File: golem/ethereum/node.py (hunk 3 above)
Comment (banasiakadam60) [reply golem/ethereum/node.py]: It looks like `etherbase address` must be set

**Review by chfast — APPROVED** [review]

**Pull request merged**